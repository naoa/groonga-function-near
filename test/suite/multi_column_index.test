register functions/near
table_create Diaries TABLE_NO_KEY
column_create Diaries title COLUMN_SCALAR ShortText
column_create Diaries body COLUMN_SCALAR ShortText

load --table Diaries
[
  {
    "title": "2014-8-12's weather",
    "body": "It was rainy in Tokyo. But it was sunny in Osaka."
  },
  {
    "title": "2014-8-13's weather",
    "body": "It was cloudy in Tokyo. But it was rainy in Sapporo."
  }
]

table_create Terms TABLE_HASH_KEY ShortText \
  --default_tokenizer TokenBigram \
  --normalizer NormalizerAuto
column_create Terms diaries_body \
  COLUMN_INDEX|WITH_POSITION|WITH_SECTION \
  Diaries title,body

select Diaries \
  --output_columns body,_score \
  --match_columns 'body' \
  --query 'tokyo' \
  --filter 'near(body, 2, "rainy tokyo")'
