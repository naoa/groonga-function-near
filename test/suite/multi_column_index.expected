register functions/near
[[0,0.0,0.0],true]
table_create Diaries TABLE_NO_KEY
[[0,0.0,0.0],true]
column_create Diaries title COLUMN_SCALAR ShortText
[[0,0.0,0.0],true]
column_create Diaries body COLUMN_SCALAR ShortText
[[0,0.0,0.0],true]
load --table Diaries
[
  {
    "title": "2014-8-12's weather",
    "body": "It was rainy in Tokyo. But it was sunny in Osaka."
  },
  {
    "title": "2014-8-13's weather",
    "body": "It was cloudy in Tokyo. But it was rainy in Sapporo."
  }
]
[[0,0.0,0.0],2]
table_create Terms TABLE_HASH_KEY ShortText   --default_tokenizer TokenBigram   --normalizer NormalizerAuto
[[0,0.0,0.0],true]
column_create Terms diaries_body   COLUMN_INDEX|WITH_POSITION|WITH_SECTION   Diaries title,body
[[0,0.0,0.0],true]
select Diaries   --output_columns body,_score   --match_columns 'body'   --query 'tokyo'   --filter 'near(body, 2, "rainy tokyo")'
[
  [
    0,
    0.0,
    0.0
  ],
  [
    [
      [
        1
      ],
      [
        [
          "body",
          "ShortText"
        ],
        [
          "_score",
          "Int32"
        ]
      ],
      [
        "It was rainy in Tokyo. But it was sunny in Osaka.",
        2
      ]
    ]
  ]
]
